service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/RespondentController.java

```diff
@@ -117,4 +149,124 @@ public void handleSubmitted(@RequestBody CallbackRequest callbackRequest) {
         }
     }
 
-}
+    @PostMapping("/change-from-other/about-to-start")
+    public AboutToStartOrSubmitCallbackResponse handleChangeFromOtherAboutToStart(
+        @RequestBody CallbackRequest callbackrequest) {
+        CaseDetails caseDetails = callbackrequest.getCaseDetails();
+        CaseData caseData = getCaseData(caseDetails);
+
+        List<Element<Other>> others = caseData.getAllOthers();
+        DynamicList otherList = othersListGenerator.buildOthersList(others);
+
+        List<String> errors = new ArrayList<>();
+        if (otherList.getListItems().isEmpty()) {
+            errors.add("There is no other person in this case.");
+        } else {
+            caseDetails.getData().put(OTHERS_LIST_KEY, otherList);
+        }
+        return respond(caseDetails, errors);
+    }
+
+    @PostMapping("/change-from-other/enter-respondent/mid-event")
+    public AboutToStartOrSubmitCallbackResponse handleChangeFromOtherEnterRespondentMidEvent(
+        @RequestBody CallbackRequest request) {
+        CaseDetails caseDetails = request.getCaseDetails();
+        CaseData caseData = getCaseData(caseDetails);
+
+        OtherToRespondentEventData eventData = caseData.getOtherToRespondentEventData();
+        Other selectedPreparedOther = othersService.getSelectedPreparedOther(caseData, eventData.getOthersList())
+            .getValue();
+        caseDetails.getData().put(TRANSFORMED_RESPONDENT,
+            respondentService.transformOtherToRespondent(selectedPreparedOther));
+
+        return respond(caseDetails);
+    }
+
+    @PostMapping("/change-from-other/mid-event")
+    public AboutToStartOrSubmitCallbackResponse handleChangeFromOtherMidEvent(
+        @RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = getCaseData(caseDetails);
+        OtherToRespondentEventData eventData = caseData.getOtherToRespondentEventData();
+
+        List<Element<Respondent>> newRespondents = confidentialDetailsService.prepareCollection(
+            caseData.getAllRespondents(), caseData.getConfidentialRespondents(), expandCollection());
+        newRespondents.add(newElement(eventData.getTransformedRespondent()));
+
+        CaseData dummyCaseData = caseData.builder()
+            .respondents1(newRespondents)
+            .build();
+
+        List<String> errors = respondentValidator.validate(dummyCaseData, dummyCaseData, true);
+        return respond(caseDetails, errors);
+    }
+
+    @PostMapping("/change-from-other/about-to-submit")
+    public AboutToStartOrSubmitCallbackResponse handleChangeFromOtherAboutToSubmit(
```

Comment:
This function's pretty long with a lot of branching statements - would be good if we could refactor/split out into helper functions